### PR TITLE
feat(NoAutoReplyMention): Inverse shift reply behaviour

### PR DESCRIPTION
**Pull request by amsyarasyiq**
Implements (#810)

### CODE REVIEW

**Comment by AutumnVN** [comment]
:FuyuStare:

File: src/plugins/noReplyMention.tsx [hunk 1]
```diff
@@ -19,40 +19,48 @@
...
+        }
     },
-    shouldMention(reply: Reply) {
-        return Settings.plugins.NoReplyMention.exemptList.includes(
-            reply.message.author.id
-        );
+    shouldMention(message: Message, holdingShift: boolean) {
+        const { exemptList, inverseShiftReply } = Settings.plugins.NoReplyMention;
+
+        const isExempted = exemptList.includes(message.author.id);
```
Comment (Vendicated) [inline src/plugins/noReplyMention.tsx:44]:
```suggestion
        const isExempt = exemptList.includes(message.author.id);
```

File: src/plugins/noReplyMention.tsx [hunk 2]
```diff
@@ -19,40 +19,48 @@
...
                 match: /CREATE_PENDING_REPLY:function\((.{1,2})\){/,
                 replace:
-                    "CREATE_PENDING_REPLY:function($1){$1.shouldMention=$self.shouldMention($1);",
+                    "CREATE_PENDING_REPLY:function($1){$1._isQuickReply&&($1.shouldMention=$self.shouldMention($1.message));",
             },
         },
+        {
+            find: ",\"Message\")}function",
+            replacement: {
+                match: /:(.{1,2}),shouldMention:!(.{1,2})\.shiftKey/,
```
Comment (Vendicated) [inline src/plugins/noReplyMention.tsx:61]:
```suggestion
                match: /:(\i),shouldMention:!(\i)\.shiftKey/,
```

File: src/plugins/noReplyMention.tsx [hunk 3]
```diff
@@ -19,40 +19,48 @@
...
     options: {
         exemptList: {
             description:
-                "List of users to exempt from this plugin (separated by commas)",
+                "List of users to exempt from this plugin (separated by commas or spaces)",
             type: OptionType.STRING,
             default: "1234567890123445,1234567890123445",
         },
+        inverseShiftReply: {
+            description: "Inverse shift replying behaviour",
```
Comment (Vendicated) [inline src/plugins/noReplyMention.tsx:36]:
```suggestion
            description: "Invert Discord's shift replying behaviour (enable to make shift reply mention user)",
```

**Comment by amsyarasyiq** [comment]
- basically rewrote the entire plugin
- new options in QuickReply--whether or not to reply ping

File: src/plugins/quickReply.ts [hunk 4]
```diff
@@ -31,10 +31,33 @@ let editIdx = -1;
 
 migratePluginSettings("QuickReply", "InteractionKeybinds");
 
+enum MentionOptions {
```
Comment (Vendicated) [inline src/plugins/quickReply.ts:34]:
```suggestion
const enum MentionOptions {
```

**Review by Vendicated — APPROVED** [review]

**Pull request merged**